### PR TITLE
[3749] Add API documentation for Subjects endpoint

### PR DESCRIPTION
**Pull request by aliuk2012**
### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by aliuk2012** [comment]
[[S] Document API endpoint - /api/public/v1/subjects](https://trello.com/c/7pKCSejg/3749-document-api-endpoint-api-public-v1-subjects)

File: spec/api/subjects_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require "swagger_helper"
+
+describe "API" do
+  path "/subjects" do
+    get "Returns a list of subjects" do
+      operationId :public_api_v1_subjects
+      tags "Subject"
```
Comment (darokel) [inline spec/api/subjects_spec.rb:7]: Should this be plural?

File: swagger/public_v1/api_spec.json [hunk 2]
```diff
@@ -1459,6 +1475,39 @@
...
+            "style": "form",
+            "explode": false,
+            "required": false,
+            "example": "subject_name",
+            "description": "Sort subjects by name"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The collection of Subjects..",
```
Comment (defong) [inline swagger/public_v1/api_spec.json:1500]: Double full stop

File: swagger/public_v1/api_spec.json [hunk 3]
```diff
@@ -1124,6 +1124,22 @@
           "code": {
             "type": "string",
             "description": "Unique subject code."
+          },
+          "bursary_amount": {
+            "type": "string",
+            "description": "non-repayable sum of money that universities can award to students to incentivise and support their studies"
```
Comment (defong) [inline swagger/public_v1/api_spec.json:1130]: Lack full stop

File: app/controllers/api/public/v1/subjects_controller.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            data: [
+              {
+                "id": "1",
+                "type": "Subject",
+                "attributes": {
+                  "subject_name": "Primary",
+                  "subject_code": "00",
+                  "bursary_amount": "14000",
+                  "early_career_payments": "2000",
+                  "scholarship": "1000",
```
Comment (defong) [inline app/controllers/api/public/v1/subjects_controller.rb:16]: should it be string vs integer?

File: swagger/public_v1/component_schemas/SubjectAttributes.yml [hunk 5]
```diff
@@ -1,12 +1,22 @@
...
+    description: "non-repayable sum of money that universities can award to students to incentivise and support their studies"
+  early_career_payments:
+    type: string
+    description: "Eligible mathematics, physics, chemistry and languages teachers can apply for early-career payments of up to £9,000 after tax."
+  scholarship:
+    type: string
+    description: "a grant or payment made to support a student’s education, awarded on the basis of academic or other achievement."
+  subject_knowledge_enhancement_course_available:
+    type: boolean
+    description: "SKE courses are available in: maths, physics, languages, chemistry, computing, biology, geography, English, religious education, design and technology, and primary maths. Schools or universities will identify your need for an SKE as part of their selection process, usually at interview. If they feel you need to enhance your knowledge, but have potential to be a great teacher, they’ll offer you a teacher training place on the condition that you complete an SKE course."
```
Comment (defong) [inline swagger/public_v1/component_schemas/SubjectAttributes.yml:22]: should we actually be stating the subjects
if it changes then this needs to change

File: swagger/public_v1/component_schemas/SubjectAttributes.yml [hunk 6]
```diff
@@ -1,12 +1,22 @@
...
     description: "Subject name."
   code:
     type: string
     description: "Unique subject code."
+  bursary_amount:
+    type: string
+    description: "non-repayable sum of money that universities can award to students to incentivise and support their studies"
+  early_career_payments:
+    type: string
+    description: "Eligible mathematics, physics, chemistry and languages teachers can apply for early-career payments of up to £9,000 after tax."
```
Comment (defong) [inline swagger/public_v1/component_schemas/SubjectAttributes.yml:16]: should we actually be stating the subjects
if it changes then this needs to change

File: app/controllers/api/public/v1/subjects_controller.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      class SubjectsController < API::Public::V1::ApplicationController
+        def index
+          render json: {
+            data: [
+              {
+                "id": "1",
+                "type": "Subject",
+                "attributes": {
+                  "subject_name": "Primary",
+                  "subject_code": "00",
```
Comment (defong) [inline app/controllers/api/public/v1/subjects_controller.rb:13]: in yml they are with out prefix of `subject_`

image

File: spec/api/subjects_spec.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+      produces "application/json"
+      parameter name: :sort,
+                in: :query,
+                schema: { "type" => "string" },
+                type: :string,
+                style: :form,
+                explode: false,
+                required: false,
+                example: "subject_name",
```
Comment (defong) [inline spec/api/subjects_spec.rb:16]:
```suggestion
                example: "name",
```

File: spec/api/subjects_spec.rb [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+                description: "Sort subjects by name"
+
+      response "200", "The list of subjects" do
+        let(:sort) { "subject_name" }
```
Comment (defong) [inline spec/api/subjects_spec.rb:20]:
```suggestion
        let(:sort) { "name" }
```

File: swagger/public_v1/component_schemas/SubjectAttributes.yml [hunk 10]
```diff
@@ -1,12 +1,25 @@
...
 properties:
   name:
     type: string
     description: "Subject name."
   code:
     type: string
     description: "Unique subject code."
+  bursary_amount:
+    type: integer
+    description: "non-repayable sum of money that universities can award to students to incentivise and support their studies"
```
Comment (defong) [inline swagger/public_v1/component_schemas/SubjectAttributes.yml:13]:
```suggestion
    description: "Non-repayable sum of money that universities can award to students to incentivise and support their studies."
```

File: swagger/public_v1/component_schemas/SubjectAttributes.yml [hunk 11]
```diff
@@ -1,12 +1,25 @@
...
+    example: 9000
+  early_career_payments:
+    type: integer
+    description: "eligible subjects teachers can apply for early-career payments of up to £9,000 after tax."
```
Comment (defong) [inline swagger/public_v1/component_schemas/SubjectAttributes.yml:17]:
```suggestion
    description: "Eligible subjects that teachers can apply for early-career payments of up to £9,000 after tax."
```

File: swagger/public_v1/component_schemas/SubjectAttributes.yml [hunk 12]
```diff
@@ -1,12 +1,25 @@
...
+    example: 8388
+  scholarship:
+    type: integer
+    description: "a grant or payment made to support a student’s education, awarded on the basis of academic or other achievement."
```
Comment (defong) [inline swagger/public_v1/component_schemas/SubjectAttributes.yml:21]:
```suggestion
    description: "A grant or payment made to support a student’s education, awarded on the basis of academic or other achievement."
```

**Pull request merged**